src/plugins/flattenModifier.ts

```diff
@@ -52,8 +68,167 @@ export class FlattenModifier {
+                        schema.addProperty(new Property(childProperty.language.default.name, childProperty.language.default.description, childProperty.schema, {
+                            // eslint-disable-next-line @typescript-eslint/no-explicit-any
+                            ...(<any>childProperty),
+                            flattenedNames: [property.serializedName, ...childProperty.flattenedNames ? childProperty.flattenedNames : [childProperty.serializedName]],
+                            required: property.required && childProperty.required
+                        }));
+                    }
+
+                    if (length(property.extensions) === 0) {
+                        delete property['extensions'];
+                    }
+                    // and mark the child class as 'do-not-generate' ?
+                    (property.schema.extensions = property.schema.extensions || {})[hasBeenFlattened] = true;
+                    NodeCliHelper.setCliFlattened(property, true);
+                }
+            }
+        }
+
+        schema.extensions[isCurrentlyFlattening] = false;
+    }
+
+    private flattenPayload(flattenMultiPayload: boolean, propNumThreshold: number): void {
+        for (const group of this.session.model.operationGroups) {
+            for (const operation of group.operations) {
+                // when there are multiple requests in an operation
+                // and the generator asks not to flatten them
+                if (length(operation.requests) > 1 && !flattenMultiPayload) {
+                    continue;
+                }
+
+                for (const request of values(operation.requests)) {
+                    const body = values(request.parameters).first(p => p.protocol.http?.in === ParameterLocation.Body && p.implementation === ImplementationLocation.Method);
+
+                    if (body && isObjectSchema(body.schema)) {
+                        const schema = <ObjectSchema>body.schema;
+                        if (schema.discriminator) {
+                            // skip flattening on polymorphic payloads, since you don't know the actual type.
+                            continue;
+                        }
+
+                        let flattenOperationPayload = NodeCliHelper.isCliFlatten(body) && !NodeCliHelper.isCliFlattened(body);
+                        if (!flattenOperationPayload) {
+                            // told not to explicitly.
+                            continue;
+                        }
+
+
+                        if (!flattenOperationPayload) {
+                            // get the count of the (non-readonly) properties in the schema
+                            flattenOperationPayload = length(values(getAllProperties(schema)).where(property => property.readOnly !== true && property.schema.type !== SchemaType.Constant)) <= propNumThreshold;
+                        }
```

Comment:
this will never be executed isn't it?

Reply:
Yes, it's a bug. I will remove the `if` statement